src/Sylius/Bundle/CoreBundle/Validator/Constraints/HasEnabledEntityValidator.php

```diff
@@ -0,0 +1,139 @@
+{
+    /**
+     * @var ManagerRegistry
+     */
+    protected $registry;
+    /**
+     * @var \Symfony\Component\PropertyAccess\PropertyAccessor
+     */
+    protected $accessor;
+
+    /**
+     * @param ManagerRegistry $registry
+     */
+    public function __construct(ManagerRegistry $registry)
+    {
+        $this->registry = $registry;
+        $this->accessor = PropertyAccess::createPropertyAccessor();
+    }
+
+    /**
+     * @param object $entity
+     * @param Constraint $constraint
+     *
+     * @throws UnexpectedTypeException
+     * @throws ConstraintDefinitionException
+     */
+    public function validate($entity, Constraint $constraint)
+    {
+        if (!$constraint instanceof HasEnabledEntity) {
+            throw new UnexpectedTypeException($constraint, __NAMESPACE__ . '\HasEnabledEntity');
+        }
+
+        $enabled = $this->accessor->getValue($entity, $constraint->enabledPath);
+
+        if ($enabled === true) {
+            return;
+        }
+
+        $entityManager = $this->getProperEntityManager($entity, $constraint);
+
+        $this->ensureEntityHasProvidedEnabledField($entity, $constraint, $entityManager);
+
+        $criteria = array($constraint->enabledPath => true);
+
+        $repository = $entityManager->getRepository(get_class($entity));
+        $result = $repository->{$constraint->repositoryMethod}($criteria);
+
+        /* If the result is a MongoCursor, it must be advanced to the first
+         * element. Rewinding should have no ill effect if $result is another
+         * iterator implementation.
+         */
+        if ($result instanceof \Iterator) {
+            $result->rewind();
+        } elseif (is_array($result)) {
+            reset($result);
+        }
+
+        /* If no entity matched the query criteria or a single entity matched,
+         * which is the same as the entity being validated, the criteria is
+         * unique.
+         */
+        if (!$result || 0 === count($result)
+            || (1 === count($result) && $entity === ($result instanceof \Iterator ? $result->current() : current(
+                    $result
+                )))
+        ) {
+            $errorPath = null !== $constraint->errorPath ? $constraint->errorPath : $constraint->enabledPath;
+
+            $this->context->addViolationAt($errorPath, $constraint->message);
+        }
+    }
+
+    /**
+     * @param $entity
+     * @param Constraint $constraint
+     * @return ObjectManager|null
+     */
+    private function getProperEntityManager($entity, Constraint $constraint)
+    {
+        if ($constraint->entityManager) {
+            $entityManager = $this->registry->getManager($constraint->entityManager);
+
+            if (!$entityManager) {
+                throw new ConstraintDefinitionException(
+                    sprintf('Object manager "%s" does not exist.', $constraint->entityManager)
+                );
+            }
+        } else {
```

Comment:
I disagree, is not boring, any comment is useful for the project and for improve the skills of people who read them.